cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -444,9 +448,29 @@ public Storage getStorage() {
     }
 
     protected void setStorage(Storage storage) {
+        if (storage instanceof PersistentClaimStorage) {
+            PersistentClaimStorage persistentClaimStorage = (PersistentClaimStorage) storage;
+            checkPersistentStorageSize(persistentClaimStorage);
+        } else if (storage instanceof JbodStorage)  {
+            JbodStorage jbodStorage = (JbodStorage) storage;
+
+            for (Storage jbodVolume : jbodStorage.getVolumes()) {
+                if (jbodVolume instanceof PersistentClaimStorage) {
+                    PersistentClaimStorage persistentClaimStorage = (PersistentClaimStorage) jbodVolume;
+                    checkPersistentStorageSize(persistentClaimStorage);
+                }
+            }
```

Comment:
You would still end up with very similar code, just moved to a different method. Why do you think it would be better there then here? I think it might be more useful here if we need to add other checks in the future, we would not need to go through the JBOD volumes multiple times.

Reply:
I proposed to move in the checkPersistentStorageSize because what we are doing here is actually just calling .... check on the storage(s).
Maybe if we have more check on storage to do, we could have checkPersistentStorageSize renamed to checkPersistentStorage which contains those checks which for now is only size.